src/Arcus.Security.Secrets.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -67,6 +68,20 @@ public KeyVaultSecretProvider(IKeyVaultAuthentication authentication, IKeyVaultC
         /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
         /// <exception cref="KeyVaultErrorException">The call for a secret resulted in an invalid response</exception>
         public async Task<string> Get(string secretName)
+        {
+            Secret secret = await GetSecret(secretName);
+            return secret?.Value;
+        }
+
+        /// <summary>
+        /// Retrieves the secret value, based on the given name
+        /// </summary>
+        /// <param name="secretName">The name of the secret key</param>
+        /// <returns>Returns a <see cref="Secret"/> that contains the secret key</returns>
+        /// <exception cref="ArgumentException">The name must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The name must not be null</exception>
+        /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        public async Task<Secret> GetSecret(string secretName)
```

Comment:
Wondering if we should return Secret or a monad.

Reply:
We're propably talking about different stuff here when we say "monad" (because `Task` is a monad in category theory/functional programming).

Reply:
Probably. I'm talking about the Maybe option
https://mikhail.io/2016/01/monads-explained-in-csharp/

Reply:
Ah, ok; than we're talking about the same thing.

Reply:
The task is a monad, but not the object that it returns. Anyway, let's skip it for now then.